src/vs/workbench/contrib/notebook/browser/view/cellParts/cellComments.ts

```diff
@@ -45,21 +44,17 @@ export class CellComments extends CellContentPart {
 	}
 
 	private async initialize(element: ICellViewModel) {
-		if (this._initialized) {
+		if (this.currentElement === element) {
 			return;
 		}
 
-		this._initialized = true;
-		const info = await this._getCommentThreadForCell(element);
-
-		if (info) {
-			await this._createCommentTheadWidget(info.owner, info.thread);
-		}
+		this.currentElement = element as CodeCellViewModel;
+		this._updateThread();
```

Comment:
Thanks for checking.

I think at the moment it does not change anything, because the call to `initialize()` in line 160 is not awaited (neither before nor after this change). The API of `didRenderCell` does not allow returning a promise.  

I could either remove the "async" from initialize (or inline the method altogether since it is only called from that one place), or could await here. 

Do you have a preference?

Reply:
I would `await` here since that should help if a future refactor causes `initialize` to be called from somewhere where it can be awaited.

Reply:
Sounds good. Done. Please take another look.